### DISCUSSION TITLE
Quantum Child Specification Error in Elixir Application

### DISCUSSION THREAD

**Discussion by obs-gh-jaydave** [quantum-elixir/quantum-core#606] [Q&A] [answered]
Hi, 

version:  `{:quantum, "~> 3.3"}`

I've encountered an issue with integrating the Quantum scheduler into my Elixir application's supervision tree. Despite following the documentation and ensuring that Quantum is correctly added as a child to the supervisor, I'm still facing an error related to the `child_spec/1 function`.

Here's the error message I'm receiving:

```
Application featureflagservice exited: exited in: Featureflagservice.Application.start(:normal, [])
** (EXIT) an exception was raised:
    ** (ArgumentError) The module Quantum was given as a child to a supervisor
    but it does not implement child_spec/1.
```

This error persists even after specifying Quantum in the application's supervision tree as follows:

```
children = [
  # ... other children ...
  {Quantum, [name: Featureflagservice.Scheduler]}
]
```

And here's the relevant part of my config.exs where I define the Quantum jobs:

```
config :featureflagservice, Featureflagservice.Scheduler,
  jobs: [
    # Jobs configuration
  ]
```

I've ensured that Quantum is included in my mix.exs dependencies and fetched with `mix deps.get`. My Elixir and OTP versions are compatible with the Quantum version I'm using.

Could anyone provide insights into what might be causing this issue or suggest a solution? Any help would be greatly appreciated.

Thank you!

**Comment by maennchen** [comment, marked as answer]
Have a look at the setup docs: https://hexdocs.pm/quantum/readme.html#setup

You should start your own scheduler and not the quantum module directly.
